zarr/core.py

```diff
@@ -1704,10 +1733,12 @@ def _chunk_setitem(self, chunk_coords, chunk_selection, value, fields=None):
                                        fields=fields)
 
     def _chunk_setitem_nosync(self, chunk_coords, chunk_selection, value, fields=None):
-
-        # obtain key for chunk storage
         ckey = self._chunk_key(chunk_coords)
+        cdata = self._process_for_setitem(ckey, chunk_selection, value, fields=fields)
+        # store
+        self.chunk_store[ckey] = cdata
 
+    def _process_for_setitem(self, ckey, chunk_selection, value, fields=None):
```

Comment:
```suggestion
    def _process_for_setitem(self, ckey, chunk_selection, value, *, fields):
```

Maybe ? You seem to alway pass fields, so no need to give it a default value ?
I leave it kwarg only, but we can remove the `*`.

Reply:
I think I'll leave it, since it's not a parameter normally expected to have a useful value, and this more closely matches the other signatures in the class.